packages/docusaurus-mdx-loader/src/remark/toc/__tests__/index.test.ts

```diff
@@ -17,14 +17,7 @@ const processFixture = async (name: string) => {
 
   const filePath = path.join(__dirname, '__fixtures__', `${name}.md`);
   const file = await vfile.read(filePath);
-  const result = await remark()
-    .use(headings)
-    .use(gfm)
-    .use(mdx)
-    .use(plugin)
-    .process(file);
-
-  return result.value;
+  return remark().use(headings).use(gfm).use(mdx).use(plugin).parse(file);
```

Comment:
This change makes snapshots much more verbose and harder to review.

Also, since snapshot format has changed, we now can't easily know if there is a TOC regression since this also updates all former tests snapshots. Keeping the former test snapshots intact is what guarantees your logic does not introduce any regression.

If you really need to capture AST in snapshots, please explain why, and do this only for the newly added tests so that we can still be sure there's no toc regression in this PR.